### PR TITLE
Added new Page for the PermissionAPI

### PR DESCRIPTION
**Pull request by JTK222**
Added a new Page for the mostly unknown PermissionAPI forge provides.
Might contain a few typos and or grammar errors, would be nice if someone could take a look at it.

### CODE REVIEW

File: docs/utilities/permissionapi.md [hunk 1]
```diff
@@ -0,0 +1,55 @@
+PermissionAPI
+=============
+
+The PermissionAPI is a very basic implementation of a permission system.
+
+It's default implementation doesn't add propper permission handling though, but instead just has 3 "permission levels", (ALL = all players, OP = operators, NONE = no player).
```
Comment (Poke1650) [inline docs/utilities/permissionapi.md]: "It's" should be "Its" here

Comment (Poke1650) [inline docs/utilities/permissionapi.md]: "proper" only takes one "p"

File: docs/utilities/permissionapi.md [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+though in case that there is no mod changing the default behaviour this will always return false.
+
+So if we want that all players, or just OP's are able to use this as default, we also need to register our permission node.
+
+Achieving this is as simple as checking for permissions: `PermissionAPI.registerNode(String node, DefaultPermissionLevel level, String description)`
+
+though this has to be done in Init or Later.
+
+!!! note
+    The PermissionAPI isn't restricted to be used for commands, you could also use it for other things, like restricting acces to a GUI.
```
Comment (Poke1650) [inline docs/utilities/permissionapi.md]: "access" takes two "s"

File: docs/utilities/permissionapi.md [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+DefaultPermissionLevel
+--------------
+
+The DefaultPermissionLevel is used by the default implementation of the IPermissionHandler, it supports 3 Values:
+`ALL = all players, OP = only operators, NONE = neither normal player or operators`
+
+Permission node
+---------------------------------------
+
+While there are theoraticly no rules for the permission nodes, the best practice for them is to be `modid.subgroup.permission_id`
```
Comment (Poke1650) [inline docs/utilities/permissionapi.md]: "theoretically" misspelled

File: docs/utilities/permissionapi.md [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+
+It is recommended to use this naming scheme as other implemntations may have stricter rules.
```
Comment (Poke1650) [inline docs/utilities/permissionapi.md]: "implementations" missing a "e"

File: docs/utilities/permissionapi.md [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+
+Making an own Implementation of the PermissionHandler
+--------------------------------------
+
+By default the PermissionHandler isn't very usefull, but this can be changed with an own instance of the IPermissionHandler.
```
Comment (Poke1650) [inline docs/utilities/permissionapi.md]: "useful" only takes one "l"

File: docs/utilities/permissionapi.md [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+
+!!! note
+    Not every mod that want's to use the PermissionAPI should change the PermissionHandler as there can be just 1 at the same time!
+
+First of how you want to implement an own PermissionHandler is completly up to you, you can use files, a database or whatever you want.
```
Comment (Poke1650) [inline docs/utilities/permissionapi.md]: "First off"?

Comment (Poke1650) [inline docs/utilities/permissionapi.md]: > you want to implement an own PermissionHandler
Sounds kinda weird to me, maybe "you want to implement your own ..." ?

File: docs/utilities/permissionapi.md (hunk 1 above)
Comment (LexManos) [inline docs/utilities/permissionapi.md]: Ya... your pessimistic "the default implementation is shit" attitude isn't gonna fly. Its simplistic for a reason. Rewrite this and the several other lines like it to be less subjective.

Comment (JTK222) [reply docs/utilities/permissionapi.md]: Oh, that wasn't my intention in the first place,
I just wrote it like that as most people know Permissions as something like you have with PEX,
and I didn't want to create any false assumptions about this. Hope the latest commit fixes that.

File: docs/utilities/permissionapi.md [hunk 7]
```diff
@@ -0,0 +1,63 @@
+PermissionAPI
+=============
+
+The PermissionAPI is a very basic implementation of a permission system.
+
+Its default implementation doesn't add an advanced permission handling (like we know it for example from PEX), 
+
+but instead it has 3 permission levels, (ALL = all players, OP = operators, NONE = neither normal players or operators).
+
+This behaviour can be changed by mods which implement an own PermissionHandler.
```
Comment (tterrag1098) [inline docs/utilities/permissionapi.md]: an -> their

File: docs/utilities/permissionapi.md [hunk 8]
```diff
@@ -0,0 +1,63 @@
...
+
+So if we want that all players, or just OP's to be able to use this  we also need to register our permission node.
+
+Achieving this is as simple as checking for permissions: `PermissionAPI.registerNode(String node, DefaultPermissionLevel level, String description)`, 
+
+though this has to be done in Init or Later.
+
+!!! note
+    The PermissionAPI isn't restricted to be used for commands, you could also use it for other things, like restricting access to a GUI.
+    You do also need to check if your CommandSender is a player if you use it in combination with commands!
```
Comment (tterrag1098) [inline docs/utilities/permissionapi.md]: You do also need -> Also, you need
CommandSender -> `ICommandSender`

File: docs/utilities/permissionapi.md [hunk 9]
```diff
@@ -0,0 +1,63 @@
...
+
+DefaultPermissionLevel
+--------------
+
+The DefaultPermissionLevel has 3 Values:
+
+Permission node
+---------------------------------------
+
+While there are theoretically no rules for the permission nodes, the best practice for them is to be `modid.subgroup.permission_id`
```
Comment (tterrag1098) [inline docs/utilities/permissionapi.md]: theoretically -> technically

Comment (tterrag1098) [reply docs/utilities/permissionapi.md]: the best practice for them is to be -> the best practice is to be of the form

File: docs/utilities/permissionapi.md [hunk 10]
```diff
@@ -0,0 +1,63 @@
...
+
+It is recommended to use this naming scheme as other implementations may have stricter rules.
+
+Making an own Implementation of the PermissionHandler
+--------------------------------------
+
+By default the PermissionHandler is very basic, which is usually enough for most users, 
+
+but you might want more controll over the permissions for things like a big server.
```
Comment (tterrag1098) [inline docs/utilities/permissionapi.md]: controll -> control

File: docs/utilities/permissionapi.md [hunk 11]
```diff
@@ -0,0 +1,63 @@
...
+
+ This can be achieved  by creating an own PermissionHandler, though it is up to you what it will be capable of,
```
Comment (tterrag1098) [inline docs/utilities/permissionapi.md]: an own -> a custom

Comment (tterrag1098) [reply docs/utilities/permissionapi.md]: This sentence runs on for far too long, break it up into two.

File: docs/utilities/permissionapi.md [hunk 12]
```diff
@@ -0,0 +1,63 @@
...
+ 
+ you can for example make a file to set permissions per player, or make it as advanced as PEX having database support.
+
+!!! note
+    Not every mod that want's to use the PermissionAPI should change the PermissionHandler as there can be just 1 at the same time!
```
Comment (tterrag1098) [inline docs/utilities/permissionapi.md]: want's -> wants
just -> only

File: docs/utilities/permissionapi.md [hunk 13]
```diff
@@ -0,0 +1,63 @@
...
+The DefaultPermissionLevel has 3 Values:
+
+Permission node
+---------------------------------------
+
+While there are theoretically no rules for the permission nodes, the best practice for them is to be `modid.subgroup.permission_id`
+
+It is recommended to use this naming scheme as other implementations may have stricter rules.
+
+Making an own Implementation of the PermissionHandler
```
Comment (tterrag1098) [inline docs/utilities/permissionapi.md]: an own -> you rown
Implementation -> implementation
PermissionHandler -> `PermissionHandler`

File: docs/utilities/permissionapi.md [hunk 14]
```diff
@@ -0,0 +1,63 @@
...
+though this has to be done in Init or Later.
+
+!!! note
+    The PermissionAPI isn't restricted to be used for commands, you could also use it for other things, like restricting access to a GUI.
+    You do also need to check if your CommandSender is a player if you use it in combination with commands!
+
+DefaultPermissionLevel
+--------------
+
+The DefaultPermissionLevel has 3 Values:
```
Comment (tterrag1098) [inline docs/utilities/permissionapi.md]: DefaultPermissionLevel -> `DefaultPermissionLevel

File: docs/utilities/permissionapi.md [hunk 15]
```diff
@@ -0,0 +1,63 @@
...
+but you might want more controll over the permissions for things like a big server.
+
+ This can be achieved  by creating an own PermissionHandler, though it is up to you what it will be capable of,
+ 
+ you can for example make a file to set permissions per player, or make it as advanced as PEX having database support.
+
+!!! note
+    Not every mod that want's to use the PermissionAPI should change the PermissionHandler as there can be just 1 at the same time!
+
+First off, how you implement your own PermissionHandler is completly up to you, you can use files, a database or whatever you want.
```
Comment (tterrag1098) [inline docs/utilities/permissionapi.md]: completly ->completely

File: docs/utilities/permissionapi.md [hunk 16]
```diff
@@ -0,0 +1,63 @@
...
+
+All you need to do is create an own instance of the interface `IPermissionHandler` and overwriting it's methods,
```
Comment (tterrag1098) [inline docs/utilities/permissionapi.md]: an own -> your own
instance of the interface -> implementation of
~~and overwriting it's methods~~

File: docs/utilities/permissionapi.md [hunk 17]
```diff
@@ -0,0 +1,63 @@
...
+
+after that is done, you also need to register it, this is done by calling `PermissionAPI.setPermissionHandler(IPermissionHandler handler)`
```
Comment (tterrag1098) [inline docs/utilities/permissionapi.md]: Make this its own sentence as well

**Comment by JTK222** [comment]
Hopefully fixed now,
and I used the OreDictionary page as an example, that's why I had the double new lines.

**Comment by JTK222** [comment]
May I ask if there are still some thing that need to be changed?
That's my first time contributing here, and the first time I am doing something in general.
So sorry if I've overseen something.

File: docs/utilities/permissionapi.md (hunk 12 above)
Comment (tterrag1098) [reply docs/utilities/permissionapi.md]: This still needs to be fixed

**Comment by tterrag1098** [comment]
I don't understand why github is hiding that comment, when the typo was not fixed. Fix that last typo and I'll merge.

**Comment by ProxyNeko** [comment]
What about this?
`+Making an own Implementation of the PermissionHandler`
an -> your

**Pull request merged**